src/modules/roc_netio/target_uv/roc_netio/udp_receiver_port.cpp

```diff
@@ -263,5 +275,39 @@ void UDPReceiverPort::recv_cb_(uv_udp_t* handle,
     self.writer_.write(pp);
 }
 
+bool UDPReceiverPort::join_multicast_group_() {
+    if (!address_.miface()) {
```

Comment:
So the real condition is `address_.multicast() && address_.miface()`. I'd keep these two together. Maybe move all checks inside join_multicast_group_()/leave_multicast_group_()?

Reply:
I would prefer to keep the both checks on the upper layer. Having unconditional `join_multicast_group_()` seems a little bit odd.

```cpp
    if (address_.multicast() && address_.miface()) {
        if (!join_multicast_group_()) {
            return false;
        }
    }
```

Reply:
Also good.